File: tam/constants/kigali-destinations.ts
/** District / city entries that count as Kigali City for passenger pricing & moto service area */
export const KIGALI_DESTINATION_IDS = new Set([
  'kigali-city',
  'gasabo',
  'kicukiro',
  'nyarugenge',
]);

export function isKigaliDestination(destinationId: string): boolean {
  return KIGALI_DESTINATION_IDS.has(destinationId);
}

/**
 * Approximate bounding box for Kigali City (taxi moto pickup zone).
 * Used with GPS so passengers outside this area cannot select taxi moto.
 */
const KIGALI_CITY_BBOX = {
  /** Southern edge (more negative latitude) */
  minLat: -2.06,
  /** Northern edge */
  maxLat: -1.87,
  minLng: 29.95,
  maxLng: 30.22,
} as const;

export function isCoordinateInKigaliCity(latitude: number, longitude: number): boolean {
  if (!Number.isFinite(latitude) || !Number.isFinite(longitude)) return false;
  return (
    latitude >= KIGALI_CITY_BBOX.minLat &&
    latitude <= KIGALI_CITY_BBOX.maxLat &&
    longitude >= KIGALI_CITY_BBOX.minLng &&
    longitude <= KIGALI_CITY_BBOX.maxLng
  );
}

File: flutter_app/lib/domain/kigali_destinations.dart
/// Parity with [tam/constants/kigali-destinations.ts]
const kigaliDestinationIds = {
  'kigali-city',
  'gasabo',
  'kicukiro',
  'nyarugenge',
};

bool isKigaliDestination(String destinationId) =>
    kigaliDestinationIds.contains(destinationId);

/// Approximate bbox for Kigali City — taxi moto pickup zone.
const double kigaliMinLat = -2.06;
const double kigaliMaxLat = -1.87;
const double kigaliMinLng = 29.95;
const double kigaliMaxLng = 30.22;

bool isCoordinateInKigaliCity(double latitude, double longitude) {
  if (!latitude.isFinite || !longitude.isFinite) return false;
  return latitude >= kigaliMinLat &&
      latitude <= kigaliMaxLat &&
      longitude >= kigaliMinLng &&
      longitude <= kigaliMaxLng;
}
